src/rootAstNodesBuilder.ts

```diff
@@ -11,33 +11,167 @@ import Lotto from "./nodes/composite/lotto";
 import Selector from "./nodes/composite/selector";
 import Sequence from "./nodes/composite/sequence";
 import Parallel from "./nodes/composite/parallel";
+import Node from "./nodes/node";
 import While from "./attributes/guards/while";
 import Until from "./attributes/guards/until";
 import Entry from "./attributes/callbacks/entry";
 import Exit from "./attributes/callbacks/exit";
 import Step from "./attributes/callbacks/step";
+import Callback from "./attributes/callbacks/callback";
+import Guard from "./attributes/guards/guard";
+import Attribute from "./attributes/attribute";
+import Composite from "./nodes/composite/composite";
+import Decorator from "./nodes/decorator/decorator";
+import Leaf from "./nodes/leaf/leaf";
+
+export type Argument<T> = {
```

Comment:
There's a big wall of types at the top of this file. I think part of that is because this file had a lot of complexity/length to it anyway.

It could be helpful to split stuff about Arguments and/or Attributes into their own files, as I think both concepts have enough types and logic.

Reply:
I have actually done this as part of my changes to add a delta time for wait nodes.